src/accelerate/test_utils/testing.py

```diff
@@ -255,7 +255,15 @@ def require_bnb(test_case):
     """
     Decorator marking a test that requires bitsandbytes. These tests are skipped when they are not.
     """
-    return unittest.skipUnless(is_bnb_available(), "test requires the bitsandbytes library")(test_case)
+    if is_bnb_available():
+        try:
+            import pytest
+
+            return pytest.mark.bnb(test_case)
+        except ImportError:
+            return test_case
+    else:
+        return unittest.skip("test requires the bitsandbytes library")(test_case)
```

Comment:
Wouldn't that throw an exception if someone would run the test suite without pytest present? I mean this whole discussion is a bit weird, I think we should just commit to pytest fully and that's it, but if we want to support people running without pytest as a runner, then we gotta leave it as is, imo...

Actually in Transformers I also bumped into a bunch of pytest things that would throw an exception if the test suite wouldn't be run with pytest, but seems noone does that anyways..

Reply:
Our test suite is pytest + unittest. It's required in our test requirements in the .py:

```
extras["test_prod"] = ["pytest>=7.2.0,<=8.0.0", "pytest-xdist", "pytest-subtests", "parameterized"]
```

And correct, we only run our tests inside `unittest.TestCases` :) (Explicit pytest tests are reserved in accelerate only in `/src/accelerate/test_utils/scripts`

Reply:
Ok, so if we assume pytest is a hard requirement, then we don't need `is_pytest_available()`, right? We could just go ahead and import it. In case we assume someone might need to run these tests without pytest, then I think we need the code unmodified as I provided it (because in your suggestion `mark` would then be undefined, right?), unless I misunderstand sth?

Reply:
It's a hard requirement, but this code exists in the base project. So it's still a test requirement and we need the check at the top :)

Reply:
Maybe I misunderstand your suggestion, Zach, but in case `pytest` is not installed, `mark` would be undefined, right? This would not lead to an error as long as `require_bnb` is only called inside of tests, which require `pytest` as a runner. But would this not possibly be detected as a linting error, as the linter cannot know this?

Reply:
Yes. Agreed (was chatting offline with Titus about it).

I think it'd be fine if we did:

```python
if bnb_available() and is_pytest_available():
   ...
return unittest....
```

Honestly it could be even simpler and we abstract out the pytest + unittest logic